SerialPrograms/Source/PokemonSV/Programs/AutoStory/PokemonSV_AutoStoryTools.cpp

```diff
@@ -0,0 +1,666 @@
+/*  AutoStoryTools
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "CommonFramework/GlobalSettingsPanel.h"
+#include "CommonFramework/Exceptions/FatalProgramException.h"
+#include "CommonFramework/Exceptions/OperationFailedException.h"
+#include "CommonFramework/InferenceInfra/InferenceRoutines.h"
+#include "CommonFramework/Notifications/ProgramNotifications.h"
+#include "CommonFramework/Tools/StatsTracking.h"
+#include "CommonFramework/ImageTools/SolidColorTest.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_PushButtons.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_Superscalar.h"
+#include "NintendoSwitch/Programs/NintendoSwitch_SnapshotDumper.h"
+#include "PokemonSwSh/Inference/PokemonSwSh_IvJudgeReader.h"
+#include "PokemonSV/Inference/Battles/PokemonSV_NormalBattleMenus.h"
+#include "PokemonSV/Inference/Dialogs/PokemonSV_DialogDetector.h"
+#include "PokemonSV/Inference/Overworld/PokemonSV_OverworldDetector.h"
+#include "PokemonSV/Inference/PokemonSV_MainMenuDetector.h"
+#include "PokemonSV/Inference/Map/PokemonSV_MapMenuDetector.h"
+#include "PokemonSV/Programs/PokemonSV_Navigation.h"
+#include "PokemonSV/Programs/PokemonSV_GameEntry.h"
+#include "PokemonSV/Programs/PokemonSV_SaveGame.h"
+#include "PokemonSV/Programs/Battles/PokemonSV_Battles.h"
+#include "PokemonSV/Programs/AutoStory/PokemonSV_MenuOption.h"
+#include "PokemonSV/Inference/PokemonSV_TutorialDetector.h"
+#include "PokemonSV/Inference/PokemonSV_PokemonMovesReader.h"
+#include "PokemonSV/Inference/Map/PokemonSV_DestinationMarkerDetector.h"
+#include "PokemonSV_AutoStoryTools.h"
+
+//#include <iostream>
+//using std::cout;
+//using std::endl;
+//#include <unordered_map>
+//#include <algorithm>
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonSV{
+
+using namespace Pokemon;
+
+
+
+
+void run_battle_press_A(
+    ConsoleHandle& console, 
+    BotBaseContext& context,
+    BattleStopCondition stop_condition
+){
+    int16_t num_times_seen_overworld = 0;
+    while (true){
+        NormalBattleMenuWatcher battle(COLOR_BLUE);
+        // SwapMenuWatcher         fainted(COLOR_PURPLE);
+        OverworldWatcher        overworld(COLOR_CYAN);
+        AdvanceDialogWatcher    dialog(COLOR_RED);
+        context.wait_for_all_requests();
+
+        int ret = wait_until(
+            console, context,
+            std::chrono::seconds(90),
+            {battle, overworld, dialog}
+        );
+        context.wait_for(std::chrono::milliseconds(100));
+
+        switch (ret){
+        case 0: // battle
+            console.log("Detected battle menu, spam first move.");
+            pbf_mash_button(context, BUTTON_A, 3 * TICKS_PER_SECOND);
```

Comment:
I assume that it should be rare that we manage to mash into the move selection, but not past it right? IIRC, we do have a detector for the move selection in case we happen to get stuck there.

Reply:
As far as I know, it shouldn't be possible to mash past the move selection, since the move animation takes so long.